Added support for quota expression

gustavo-iniguez-goya opened the pull request:
Hi!

This PR adds support for quotas.

stapelberg commented:
Thanks for your PR! Could you add a corresponding test function to `nftables_test.go` as well please?

gustavo-iniguez-goya commented:
Added! sorry, I had some problems figuring out how to extract the want-ed bytes for the tests.
While debugging it I've realized that the flags attribute must be added unconditionally so I've updated quota.go accordingly.

stapelberg commented on expr/quota.go:
```diff
@@ -0,0 +1,76 @@
+// Quota defines a threshold against a number of bytes.
+type Quota struct {
+	Bytes uint64
+	Used  uint64
```
Should this field be named `Consumed` instead? That would match terminology on the kernel side at least.

gustavo-iniguez-goya replied:
yes, you're right. I took "used" from the output of `list ruleset`.

stapelberg commented on nftables_test.go:
```diff
@@ -4768,6 +4768,77 @@ func TestNotrack(t *testing.T) {
+		Exprs: []expr.Any{
+			// [ quota bytes 6 consumed 0 flags 1 ]
+			&expr.Quota{
+				Bytes: uint64(6),
```
Here and in the next line, I don’t think you need the explicit `uint64(6)`, and just using `6` should work, no?

stapelberg commented:
Thanks!
